### DISCUSSION TITLE
Unable to use DNAS Shoutcast for  broadcasting

### DISCUSSION THREAD

**Discussion by 404Mhz** [AzuraCast/AzuraCast#4764] [Q&A]
Hello, 
I've successfully put my radio on the air with Azuracast and Shoutcast, but Shoutcast use an unsecure IP (http://86.105.212.15:8000/radio.mp3) instead of https stream (https://radio.kaed-tzo.com/radio/8000/radio.mp3). There is the message of the shoutcast team : 

_You will need to physically edit the config file in a text editor.
Some 3rd-party hosting providers also allow you to edit the config file via a control panel such as Centova.
You can only view or reload the **sc_serv.conf** configuration file from there.
But you can't actually edit it there._

_Unfortunately, I'm not personally familiar with Azuracast, so I can only recommend that you contact their support for further help & instructions._

Regards,

Shoutcast Team_

After looking several time in /var/azura/ and subdirectory, i can't found any files named file **sc_serv.com** via ftp...

Could you help me please? Thank you very much

**Comment by Vaalyn** [comment]
You can add settings to the SC configuration by visiting the `Edit Station Profile` page, going to the `Broadcasting` tab and putting the settings into the `Custom Configuration` at the bottom of the page. These settings will be merged into the generated `sc_serv.conf` when restarting the station.

**Reply by 404Mhz** [reply]
Thank you so much! The shoutcast team told me to add this to the sc_serv.conf file:

_You will need to add the following lines to your sc_serv.conf configuration file:

sslCertificateFile =
sslCertificateKeyFile =

destip = radio.kaed-tzo.com
publicip = radio.kaed-tzo.com
stream path = radio / 8000 / radio.mp3

Make sure the relative paths to the certificate and key are correct, eg.
sslCertificateFile = / path / to / your / SSL.crt_

But I don't know where to find the ssl certificates...
Thanks again for your help, and sorry again for my poor English.

**Reply by Vaalyn** [reply]
Using the example domain `example.azuracast.com` here are the paths to the certificate files, just replace the domain name with the one you are using:

Certificate path: `/etc/nginx/certs/example.azuracast.com.crt`
Key path: `/etc/nginx/certs/example.azuracast.com.key`

**Reply by 404Mhz** [reply]
I've looking for nginx folder in /etc, but I cannot found it...
There is my /etc/ in ftp : [https://u-animity.com/upload/no_nginx.png](url)
I'm so confused

**Reply by Vaalyn** [reply]
You are looking on your servers directories, but the `/etc/nginx/certs/` path that I'm referencing is inside a Docker containers volume.

**Reply by 404Mhz** [reply]
I'm very sorry, I'm totally a newbie with Azuracast.
I go to this folder, but I can't find nginx folder :/
[https://u-animity.com/upload/cant_find_nginx_folder.png](url)

**Reply by Vaalyn** [reply]
For the shoutcast config you do not have to use the path as seen from your host server but how the docker container sees it, just use the paths I mentioned above (with changing the `example.azuracast.com` in the path to `yourdomain.com`).

The volume directory on the host server in that those files reside is `/var/lib/docker/volumes/azuracast_letsencrypt/_data/` but as I said, you won't need this path.

**Reply by 404Mhz** [reply]
> 
> 
> Using the example domain `example.azuracast.com` here are the paths to the certificate files, just replace the domain name with the one you are using:
> 
> Certificate path: `/etc/nginx/certs/example.azuracast.com.crt` Key path: `/etc/nginx/certs/example.azuracast.com.key`

Ok, I just have to put (and modify with my web adresse) this in **Edit Station Profile** > **Broadcasting** > **Custom Configuration**, and that all?

**Reply by Vaalyn** [reply]
Yes, just pay attention to the required format for the text field and make sure to hit `Restart Broadcasting` in the `Utilities` afterwards to regenerate the config files.

Example:

```
{"sslCertificateFile": "/etc/nginx/certs/example.azuracast.com.crt", "sslCertificateKeyFile": "/etc/nginx/certs/example.azuracast.com.key"}
```

**Reply by 404Mhz** [reply]
Thank you! My webradio broadcast now in https. :)
Another question, the stream is from _https://86.105.212.15:8000/radio.mp3_ and not _https://radio.kaed-tzo.com/radio/8000/radio.mp3._
Firefox and others browser notice the users with this type of message : 

_Warning: Potential Security Risk Ahead message, you may: Contact the website owner and ask them to correct their certificate. Click Go Back (Recommended), or visit a different website. If you are on a corporate network or using antivirus software, reach out to the support teams for assistance._

How can I fix this?
**EDIT :** Now Shoutcast show the following message : "Unable to listen to the stream URL."

**Reply by Vaalyn** [reply]
I'm not very familiar with shoutcasts configuration and shoutcast in general since it's closed source (so I'm not really using it) but I think there might be a config value that you need to set to the domain. Maybe `destip` but I'd recommend asking their support what config value controls that.

**Reply by 404Mhz** [reply]
> 
> 
> I'm not very familiar with shoutcasts configuration and shoutcast in general since it's closed source (so I'm not really using it) but I think there might be a config value that you need to set to the domain. Maybe `destip` but I'd recommend asking their support what config value controls that.

No problems, i try this :
**{"sslCertificateFile": "/etc/nginx/certs/radio.kaed-tzo.com.crt", "sslCertificateKeyFile": "/etc/nginx/certs/radio.kaed-tzo.com.key",destip="radio.kaed-tzo.com",publicip="radio.kaed-tzo.com",streampath="radio/8000/radio.mp3"}**

But nothing work for the moment ¯\_(ツ)_/¯

**Comment by 404Mhz** [comment]
I am totally lost on the configuration, the site appears in htlm but Shoutcast returns the following address : https: //radio.kaed-tzo.com: 8000 / radio.mp3
With irrelevant ports blocking the use of the Radio GUID, if annyone can help me? I'm on debian 10
Sorry for my bad english...

**Reply by YosefAdPro** [reply]
If you want to access the port directly in SHOUTCAST with HTTPS you need to have a paid license.
Otherwise you will have to work without HTTPS.

**Reply by SpeedyCharly** [reply]
Go to a service like RCAST [https://players.rcast.net/] and set your station up in there. RCAST will provide you with free embeddable players and a working https link

**Reply by bombelman** [reply]
the whole idea behing Azuracast is to do it yourself instead of "buying" streaming services.
Also, Azuracast does support Shoutcast http and https, even without a paid license....

**Reply by SpeedyCharly** [reply]
I use free RCAST players on my web site because the native Azuracast player isn't customizable enough for various uses (popup player, full screen player, etc.). I also do stream via Shoutcast in https mode without a licence on a free certificate from LetsEncryt.  So, with a little help from free 3rd party sources, it is entirely possible to build a pro-looking station on a shoestring budget...
